src/classes/RD2_ScheduleService.cls

```diff
@@ -172,6 +172,10 @@ public without sharing class RD2_ScheduleService {
         } else if (oldRd == null) {
             isNeeded = true;
 
+        // Invalid Installment Frequency should never result in a schedule update. This is a fail safe measure only
+        } else if (newRd.InstallmentFrequency__c == null || newRd.InstallmentFrequency__c < 1 || newRd.InstallmentFrequency__c > 20) {
```

Comment:
I have reviewed code but need to test the change wrt schedule calculations and Active Schedule and Installments components.

Reply:
I have tested the Installment component and the future installments are calculated correctly.